src/durableorchestrationstatus.ts

```diff
@@ -2,16 +2,61 @@ import { OrchestrationRuntimeStatus } from "./classes";
 import * as types from "durable-functions";
 
 export class DurableOrchestrationStatus implements types.DurableOrchestrationStatus {
+    public readonly name: string;
+    public readonly instanceId: string;
+    public readonly createdTime: Date;
+    public readonly lastUpdatedTime: Date;
+    public readonly input: unknown;
+    public readonly output: unknown;
+    public readonly runtimeStatus: OrchestrationRuntimeStatus;
+    public readonly customStatus?: unknown;
+    public readonly history?: Array<unknown>;
+
     /** @hidden */
-    constructor(
-        public readonly name: string,
-        public readonly instanceId: string,
-        public readonly createdTime: Date,
-        public readonly lastUpdatedTime: Date,
-        public readonly input: unknown,
-        public readonly output: unknown,
-        public readonly runtimeStatus: OrchestrationRuntimeStatus,
-        public readonly customStatus?: unknown,
-        public readonly history?: Array<unknown>
-    ) {}
+    constructor(init: unknown) {
+        if (!this.isDurableOrchestrationStatusInit(init)) {
+            throw new TypeError(
+                `Malformed data passed to DurableOrchestrationStatus constructor. Data received: ${JSON.stringify(
+                    init
+                )}`

```

Comment:
Are we able to surface specifically what check failed in our validation check?

Reply:
We could, but that means I would need to have a separate `if` statement for each field. Is it worth it?

Reply:
Done. I modified your message to explicitly state that the initializer had invalid types or missing fields.